nmigen/hdl/ast.py

```diff
@@ -797,7 +797,7 @@ class Signal(Value, DUID):
         name this ``Signal`` is assigned to. Name collisions are automatically resolved by
         prepending names of objects that contain this ``Signal`` and by appending integer
         sequences.
-    reset : int
+    reset : int, bool or integral Enum
```

Comment:
Nit: `bool` is an integral type, so it's fine to just say `int or integral Enum`; this covers the cases where it's bool or bool Enum.

Reply:
I prefer to mention ``bool`` explicitly here as it makes clear that it is a proper use of the parameter.

Reply:
This is inconsistent with all other docstrings nMigen has, which assume that the reader knows `bool` is an integral type.

Reply:
I'll change it but it is not how I read autodocs. To me it's not about knowing if ``bool`` is an integral type but if ``int`` as a specification in docstring means one is also supposed to pass  ``bool`` for that parameter. To me just specifying ``int`` means one is not supposed use ``bool`` for that parameter.

Reply:
That's not how object-oriented programming works. If `bool` derives from `int`, then it is OK to pass `bool` anywhere you'd pass `int`; this is called the Liskov Substitution Principle. Over-explaining LSP in a docstring is like over-explaining classes in a docstring: sure, you could do it, but why?